### PR TITLE
Improve docs and samples for dates and timezones

### PR DESCRIPTION
**Pull request by ekoleda-codaio**
- Added a [guide on timezones](https://coda-packs-sdk--1773.com.readthedocs.build/en/1773/guides/advanced/timezones/).
- Added [samples](https://coda-packs-sdk--1773.com.readthedocs.build/en/1773/samples/topic/dates/) showing working with dates and times.
- Fixed some the date logic in some existing samples.
- Small updates to the date information in the parameters and data types guide.

### CODE REVIEW

File: docs/guides/advanced/timezones.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# Timezones in Coda
+
+Every Coda document has a **Timezone** setting in the **Region** section of the settings menu, that defaults to the timezone of the device that created it. All dates and times entered in the document are interpreted in the context of that timezone. Users often don't need to be aware this, since all of their interactions with the dates and times take place in the context of that doc.
+
+Packs don't have a configurable timezone however, and all date and time operations in Pack code run in [Coordinated Universal Time (UTC)][wikipedia_utc]. The timezone of the document is passed to Pack formulas as [`context.timezone`][ExecutionContext_timezone], containing an [IANA timezone identifier][wikipedia_tz] (ex: `America/New_York`).
+
+
+## Timezone shifting {: .shifting}
+
+Any date or time values passed into a Pack are shifted from the doc's timezone to UTC. For example, the value "11/12/1955, 10:04 PM" in a document with the timezone set to "America/Los Angeles" will be passed to a Pack as `1955-11-13T06:04:00.000Z`. While this accurately represents the exact same moment in time, you'll notice that the day and hour values have changed due to the change in timezone.
```
Comment (huayang-codaio) [inline docs/guides/advanced/timezones.md:14]: lol I like your example of using 1955. lol

File: docs/guides/advanced/timezones.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+You can get a string version of a date object localized to the timezone of the document by using the `toLocaleString()` method and the `timeZone` option:
+
+```ts
+let formatted = date.toLocaleString("en-US", {
+  timeZone: context.timezone,
+});
+```
+
+There are also equivalent `getLocaleDateString()` and `getLocaleTimeString()` methods for getting just the date or time component respectively.
```
Comment (huayang-codaio) [inline docs/guides/advanced/timezones.md:39]: 👍

File: docs/guides/basics/data-types.md [hunk 3]
```diff
@@ -181,7 +181,7 @@ When used with a number value, the number should contain the number of seconds s
       codaType: coda.ValueHintType.DateTime,
       execute: async function ([], context) {
         let now = new Date();
-        return now.getTime();
+        return now.getTime() / 1000;
```
Comment (huayang-codaio) [inline docs/guides/basics/data-types.md:184]: 👍

**Review by huayang-codaio — APPROVED** [review]
Thanks for adding the documents. I didn't know these date/time tricks myself. Learned today!

File: docs/guides/advanced/timezones.md (hunk 1 above)
Comment (ekoleda-codaio) [reply docs/guides/advanced/timezones.md:14]: Back to the Future reference!

**Pull request merged**